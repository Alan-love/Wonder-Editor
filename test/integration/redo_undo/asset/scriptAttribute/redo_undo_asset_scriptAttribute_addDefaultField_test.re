open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

let _ =
  describe("redo_undo: asset script attribute->add default field", () => {
    let sandbox = getSandboxDefaultVal();

    let _simulateOnceUpdateScriptAttributeData = () => {
      let assetTreeData =
        MainEditorAssetTreeTool.BuildAssetTree.buildEmptyAssetTree();
      let addedNodeId = MainEditorAssetIdTool.getNewAssetId();

      MainEditorAssetHeaderOperateNodeTool.addScriptAttribute();

      ScriptAttributeInspectorTool.addDefaultField(
        ~sandbox,
        ~nodeId=addedNodeId,
        (),
      );

      let attributeName =
        ScriptAttributeInspectorTool.getAttributeName(addedNodeId)
        |> StateLogicService.getEditorState;

      let attribute =
        ScriptAttributeInspectorTool.getAttribute(addedNodeId)
        |> StateLogicService.getEditorState;

      addedNodeId;
    };

    beforeEach(() => {
      sandbox := createSandbox();
      MainEditorSceneTool.initState(~sandbox, ());
      MainEditorSceneTool.createDefaultScene(
        sandbox,
        MainEditorSceneTool.setFirstCubeToBeCurrentSceneTreeNode,
      );
    });
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));

    describe("test undo operate", () =>
      describe("test undo one step", () =>
        test("step which from first to zero", () => {
          let addedNodeId = _simulateOnceUpdateScriptAttributeData();

          RedoUndoTool.undoHistoryState();

          BuildComponentTool.buildScriptAttributeInspectorComponent(
            ~currentNodeId=addedNodeId,
            (),
          )
          |> ReactTestTool.createSnapshotAndMatch;
        })
      )
    );

    describe("test redo operate", () =>
      describe("test redo one step", () =>
        test(
          "undo step which from first to zero, redo step which from zero to first",
          () => {
          let addedNodeId = _simulateOnceUpdateScriptAttributeData();

          RedoUndoTool.undoHistoryState();
          RedoUndoTool.redoHistoryState();

          BuildComponentTool.buildScriptAttributeInspectorComponent(
            ~currentNodeId=addedNodeId,
            (),
          )
          |> ReactTestTool.createSnapshotAndMatch;
        })
      )
    );
  });