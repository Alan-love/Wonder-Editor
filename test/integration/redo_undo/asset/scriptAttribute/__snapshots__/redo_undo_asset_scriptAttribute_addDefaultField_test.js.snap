// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redo_undo: asset script attribute->add default field test redo operate test redo one step undo step which from first to zero, redo step which from zero to first 1`] = `
<article
  className="wonder-scriptAttribute-inspector"
>
  <h1>
    Script Attribute
  </h1>
  <hr />
  <article
    className="inspector-item"
  >
    <div
      className="item-header"
      title="script attribute's name"
    >
      Name
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="New Script Attribute"
      />
    </div>
  </article>
  <div
    className="scriptAttribute-data"
  >
    <div
      className="data-fields"
    >
      <article
        className="attributeBox-component"
      >
        <div
          className="component-header"
        >
          <div
            className="header-title"
          >
            Attribute
          </div>
          <div
            className="header-close"
          >
            <img
              onClick={[Function]}
              src="./public/img/close.png"
            />
          </div>
        </div>
        <article
          className="inspector-item"
        >
          <div
            className="item-header"
            title=""
          >
            Name
          </div>
          <div
            className="item-content"
          >
            <input
              className="input-component float-input"
              onBlur={[Function]}
              onChange={[Function]}
              type="text"
              value="New Script Attribute Field"
            />
          </div>
        </article>
        <div
          className="component-content"
        >
          <article
            className="inspector-item"
          >
            <div
              className="item-header"
              title="defines how textures are sampled when a texel is insufficient to cover a single pixel"
            >
              Type
            </div>
            <div
              className="item-content"
            >
              <select
                onChange={[Function]}
                value="1"
              >
                <option
                  value="0"
                >
                  Int
                </option>
                <option
                  value="1"
                >
                  Float
                </option>
              </select>
            </div>
          </article>
        </div>
        <article
          className="inspector-item wonder-float-input"
        >
          <div
            className="item-header component-label"
            onMouseDown={[Function]}
            onMouseMove={[Function]}
            onMouseUp={[Function]}
            title=""
          >
            Value
          </div>
          <div
            className="item-content"
          >
            <input
              className="input-component float-input"
              onBlur={[Function]}
              onChange={[Function]}
              type="text"
              value="0."
            />
          </div>
        </article>
      </article>
    </div>
    <hr />
    <button
      className="data-addable-btn"
      onClick={[Function]}
    >
      Add script attribute's field
    </button>
  </div>
</article>
`;

exports[`redo_undo: asset script attribute->add default field test undo operate test undo one step step which from first to zero 1`] = `
<article
  className="wonder-scriptAttribute-inspector"
>
  <h1>
    Script Attribute
  </h1>
  <hr />
  <article
    className="inspector-item"
  >
    <div
      className="item-header"
      title="script attribute's name"
    >
      Name
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="New Script Attribute"
      />
    </div>
  </article>
  <div
    className="scriptAttribute-data"
  >
    <div
      className="data-fields"
    />
    <hr />
    <button
      className="data-addable-btn"
      onClick={[Function]}
    >
      Add script attribute's field
    </button>
  </div>
</article>
`;
